storage/heap/ha_heap.h

```diff
@@ -63,6 +63,7 @@ class ha_heap : public handler {
                 : HA_ONLY_WHOLE_INDEX | HA_KEY_SCAN_NOT_ROR);
   }
   uint max_supported_keys() const { return HP_MAX_KEY; }
+  uint max_supported_key_length() const { return HP_MAX_KEY_LENGTH; }
```

Comment:
How did this appear here? Which commit in 5.7?

Reply:
```
'storage/heap/ha_heap.h'
"uint max_supported_key_length() const { return HP_MAX_KEY_LENGTH; }"
method override in 'ha_heap' class never got to 8.0.
See "Import memory_dynamic_rows.patch"
```
(commit 51b5cb2).

Reply:
That's the original patch import to 5.5. But this being null merge suggests that this should have appeared in the 5.7 commit range being merged. ATM I cannot exclude max_supported_key_length being dropped on purpose from 8.0 on purpose, for example

Reply:
@laurynas-biveinis 
5.5, 5.6 and 5.7 branches have this have this `max_supported_key_length()` method overridden in `ha_heap` class.
The only commit I found that removes this override is your Initial "Percona Server 8.0.11 tree"
`git show 4ad1d0ccd476ae4990dac354b382d0e2d7220be8 -- storage/heap/ha_heap.h`

```diff
...
-   uint max_supported_keys() const { return MAX_KEY; }
-   uint max_supported_key_part_length() const { return MAX_KEY_LENGTH; }
 -  const key_map *keys_to_use_for_scanning() { return &btree_keys; }
 -  uint max_supported_keys()          const { return HP_MAX_KEY; }
 -  uint max_supported_key_length() const { return HP_MAX_KEY_LENGTH; }
++  uint max_supported_keys() const { return HP_MAX_KEY; }
+   uint max_supported_key_part_length() const { return HP_MAX_KEY_LENGTH; }
 -  double scan_time()
 -  { return (double) (stats.records+stats.deleted) / 20.0+10; }
 -  double read_time(uint index, uint ranges, ha_rows rows)
 -  { return (double) rows /  20.0+1; }
 +  double scan_time() {
 +    return (double)(stats.records + stats.deleted) / 20.0 + 10;
 +  }
 +  double read_time(uint, uint, ha_rows rows) { return (double)rows / 20.0 + 1; }
...
```
Unfortunately, there is nothing related to this file in the comments for this commit.

Another version of this weirdness is Oracle force-pushing 5.7.24. There were 2 orphan commits, which I reverted as part of the upstream 5.7.24 merge (https://github.com/percona/percona-server/pull/2729). One of them have changes very close to `max_supported_key_length()` method
(https://github.com/percona/percona-server/pull/2729/commits/e13e014b0f65ce4142f519a256655d6c38a8042b#diff-bb6ecd31cb792b2abb4bc7b806e7b5ff).

Reply:
The most likely reason I did this was because the parent class implementation is suitable for HEAP, ie. the key length is not restricted further. I'd leave this out but go ahead if you believe this must go back in